meshroom/core/desc.py

```diff
@@ -313,12 +313,11 @@ def __init__(self, name, label, description, value, values, exclusive, uid, grou
         super(ChoiceParam, self).__init__(name=name, label=label, description=description, value=value, uid=uid, group=group, advanced=advanced,
                                           semantic=semantic, enabled=enabled, validValue=validValue, errorMessage=errorMessage)
 
-    def conformValue(self, val):
+    def conformValue(self, value):
         """ Conform 'val' to the correct type and check for its validity """
-        val = self._valueType(val)
-        if val not in self.values:
-            raise ValueError('ChoiceParam value "{}" is not in "{}".'.format(val, str(self.values)))
-        return val
+        if not isinstance(value, self._valueType):
+            raise ValueError('ChoiceParam value should be a {} (param:{}, value:{}, type:{})'.format(self._valueType, self.name, value, type(value)))
+        return value
```

Comment:
```suggestion
        return self._valueType(value)
```

Reply:
We just need to remove the check that the value should be in the list of possible values.

Reply:
This cannot be done as such because some types cannot be cast to others (e.g. if the user provides a string for a ChoiceParam that expects integers, the cast will fail and cause errors).